From f7ea286489cee5d8868a17d9b9da245c4ad49d4a Mon Sep 17 00:00:00 2001
From: Song Bing <bing.song@nxp.com>
Date: Mon, 26 Nov 2018 16:25:29 -0800
Subject: [PATCH 16/75] v4l2videodec: return error when flush.

Signed-off-by: Song Bing <bing.song@nxp.com>
(cherry picked from commit a73374626d890cc9d3688c0e91dc7ac2994c0671)

Upstream-Status: Pending

Conflicts:
	sys/v4l2/gstv4l2videodec.c
Signed-off-by: Xiaolei Wang <xiaolei.wang@windriver.com>
---
 sys/v4l2/gstv4l2videodec.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sys/v4l2/gstv4l2videodec.c b/sys/v4l2/gstv4l2videodec.c
index 18cbb5d..7baae49 100644
--- a/sys/v4l2/gstv4l2videodec.c
+++ b/sys/v4l2/gstv4l2videodec.c
@@ -674,7 +674,8 @@ gst_v4l2_video_dec_drain (GstVideoDecoder * decoder)
 
   GST_DEBUG_OBJECT (self, "Draining...");
   gst_v4l2_video_dec_finish (decoder);
-  gst_v4l2_video_dec_flush (decoder);
+  if (!gst_v4l2_video_dec_flush (decoder))
+    return GST_FLOW_ERROR;
 
   return GST_FLOW_OK;
 }
-- 
2.43.0

